test/functional/hosts_controller_test.rb

```diff
@@ -963,6 +963,35 @@ def test_submit_multiple_rebuild_config_pessimistic
     assert_not_nil flash[:error]
   end
 
+  test '#process_hostgroup changes compute attributes' do
+    Fog.mock!
+    group1 = FactoryGirl.create(:hostgroup, :compute_profile => compute_profiles(:one))
+    host = FactoryGirl.create(:host, :managed, :on_compute_resource)
+    host.stubs(:queue_compute)
+    host.hostgroup = group1
+    host.compute_resource = compute_resources(:one)
+    host.compute_profile = compute_profiles(:one)
+    host.set_compute_attributes
+    host.save!
```

Comment:
Is the `save!` (and `.create` earlier) required?  A new/unsaved host should be equivalent to the New Host form.